My Site: fix iPad default selected row

leandroalonso opened the pull request:
Part of #17873 
Fixes #17958

This PR fixes a bug that has been there for quite some time: the default selected row when using iPad.

### To test

#### With My Site Dashboard flag disabled:

1. Do a clean install of the app on an iPad
2. Login with an account with multiple sites
3. Accept the help when prompted
4. ✅ "Stats" should be the selected row
5. Tap any other section
6. Switch to another site
7. ✅ "Stats" should be the selected row

#### With My Site Dashboard flag enabled

1. Enable My Site Dashboard feature flag
2. Close and reopen the app
3. ✅ "Stats" should be the selected row
4. Tap "Home"
5. Switch to another site
3. ✅ "Stats" should be the selected row

Make sure to check the site that has Quick Start enabled too, "Stats" should be the selected row there.

#### My Site Dashboard flag enabled and dashboard as the default

1. Create a new site (this will make the Dashboard the default)
2. ✅ "Home" should be the selected row
3. Switch to another site
2. ✅ "Home" should be the selected row

## Regression Notes
1. Potential unintended areas of impact
iPhone

4. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested on iPhone

5. What automated tests I added (or what prevented me from doing so)
-

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

leandroalonso committed:
fix: stats and activity is in the jetpack category

leandroalonso committed:
add: show dashboard or stats depending on the default

leandroalonso committed:
docs: update release notes

pachlava reviewed (approved):
From my side, it all looks good! Tested on iPad simulator, and also checked for regressions on iPhone. Thanks, @leandroalonso!

hassaanelgarem reviewed (commented):
Works as described except for one issue.

After switching from a site that has Quick Start enabled to a site that doesn't have it enabled. The shown section is "Stats" but the selected cell is "Posts"
<img src="https://user-images.githubusercontent.com/25306722/158068955-948cdc6e-35b3-46a1-b7f5-a3f7177d6975.png" width=40%>

momo-ozawa reviewed (commented):
I'm seeing a something similar to what @hassaanelgarem mentioned above -- when switching from a site that has Quick Start enabled to a P2 site, "Stats" is shown but "Menus" is selected

<img src="https://user-images.githubusercontent.com/6711616/158118680-8be89804-3c7f-4964-9074-267f28b693ff.png" width=300>

leandroalonso committed:
fix: differentiate between a site with "jetpack" and "general" section

leandroalonso committed:
fix: make sure to update the options before selecting a cell

leandroalonso commented:
@momo-ozawa @hassaanelgarem the issues you were seeing should be fixed with the last commits. Could you please try again?

leandroalonso committed:
refactor: remove @unknown to supress warning

Context: we have 16 BlogDetailsSubsection and @unknown requires the switch to be exhaustive. However, that was never the case. I removed it to suppress the warning.

leandroalonso committed:
refactor: remove log line

leandroalonso committed:
test: fix tests

hassaanelgarem reviewed (approved):
Yes, the issue is fixed for me 🚀

momo-ozawa reviewed (approved):
LGTM!
